doc/cephfs/nfs.rst

```diff
@@ -0,0 +1,157 @@
+===
+NFS
+===
+
+CephFS namespaces can be exported over NFS protocol using the
+`NFS-Ganesha NFS server <https://github.com/nfs-ganesha/nfs-ganesha/wiki>`_.
+
+Requirements
+============
+
+-  Ceph filesystem (preferably latest luminous or later versions)
+-  'nfs-ganesha' and 'nfs-ganesha-ceph' packages (preferably the latest
+   ganesha v2.5 stable or later stable versions)
+
+   +  RPMs and packages for xenial available at http://download.ceph.com/nfs-ganesha/
+
+-  NFS-Ganesha server host connected to the Ceph public network
+
+Configuring NFS-Ganesha to export CephFS
+========================================
+
+NFS-Ganesha provides a File System Abstraction Layer (FSAL) to plug in different
+storage backends. `Ceph FSAL <https://github.com/nfs-ganesha/nfs-ganesha/tree/next/src/FSAL/FSAL_CEPH>`_
+is the plugin FSAL for CephFS. For each NFS-Ganesha export, Ceph FSAL uses a
+libcephfs client, user-space CephFS client, to mount the CephFS path that
+NFS-Ganesha exports.
+
+Setting up NFS-Ganesha with CephFS, involves setting up NFS-Ganesha's
+configuration file, and also setting up a Ceph configuration file and cephx
+access credentials for the Ceph clients created by NFS-Ganesha to access
+CephFS.
+
+
+NFS-Ganesha configuration
+-------------------------
+
+A minimal ganesha.conf for configuring NFS-Ganesha over CephFS includes a
+EXPORT block and a CEPH block. Following is an example::
+
+    EXPORT
+    {
```

Comment:
I feel like this minimal configuration should also disable the Ganesha caching. Ganesha caching should be disabled 100% of the time and we should document taht.

Reply:
when we tested, we found that if nfs-ganehsa has cache, it will cause client cache pressure. Using Attr_Expiration_Time=0 does not close nfs-ganesha's cache, Is there any other parameter to turn off the nfs-ganehsa parameter?

Reply:
> I feel like this minimal configuration should also disable the Ganesha caching. Ganesha caching should be disabled 100% of the time and we should document taht.

@batrick understood.

Reply:
@renhwztetecs , did you also try disabling or minimizing Ganesha's dentry, open fd, inode caching as mentioned here,
https://review.gerrithub.io/#/c/403825/6/src/config_samples/ceph.conf
Please see line 53 to 68

Reply:
@ajarr Thank you very much for your hints.

@ajarr @batrick @jtlayton 
In addition, I think that NFS-Ganehsa lacks a forget mechanism like FUSE(calling ceph's Client::ll_forget()) to periodically release the cache mechanism in libcephfs because of the large number of files passing through NFS-Ganehsa. When writing, the "failing to respond to cache pressure" warnings alert will appear on the libcephfs side. See this link: http://tracker.ceph.com/issues/12334
Any suggestions are very grateful?

Reply:
@renhwztetecs if you have an example reproducer, please post to that issue.